app/models/event_instance.rb

```diff
@@ -0,0 +1,33 @@
+class EventInstance < ApplicationRecord
+  # Virtual attribute for the parent event's name in case it is new
+  attr_accessor :new_parent_event_name
+
+  belongs_to :event
+  has_many   :submissions
+
+  default_scope { order :year }
+
+  before_validation :set_parent_event
+
+  validates :event, presence: true
+  validates :year, uniqueness:   { scope: :event, message: 'should happen once per year' },
+                   numericality: { only_integer: true }
+
+  def name_and_year
+    "#{event.name} #{year}"
+  end
+
+  private
+
+  def set_parent_event
+    # If both an event was selected and a new event name provided, add an error
+    if event && new_parent_event_name.present?
+      errors.add(:base, 'choose either an existing event or to create a new one')
+    # If neither an event or new event name was provided, also add an error
+    elsif event.nil? && new_parent_event_name.blank?
+      errors.add(:base, 'choose either an existing event or to create a new one')
+    else
+      create_event!(name: new_parent_event_name) unless event
```

Comment:
Where has this method come from? Is it some Rails magic??

Reply:
There's probably room to refactor this method, especially so that the comments aren't needed — and this is where the Code Climate error is stemming from.

Will do some more thinking here and come back to it!

Reply:
`ActiveModel#create_association!` is added to the model when you call `belongs_to`, so definitely Rails Magic (TM)!

I'll take a swing at refactoring the branching conditional too. I'm a little hesitant to write more helpers to handle the AND/ORs, but that might be the clearest approach. Will check.

Reply:
Ooh, you learn something new every day! Cool. :-)

Re the branch conditional..maybe there's room to extract some of the branches into explanatory methods as a first step? I'd need to take a closer look to think about best approach.